kube-runtime/src/events.rs

```diff
@@ -168,13 +208,13 @@ impl From<&str> for Reporter {
 /// ```yaml
 /// - apiGroups: ["events.k8s.io"]
 ///   resources: ["events"]
-///   verbs: ["create"]
+///   verbs: ["create", "patch"]
 /// ```
 #[derive(Clone)]
 pub struct Recorder {
-    events: Api<K8sEvent>,
+    client: Client,
     reporter: Reporter,
-    reference: ObjectReference,
+    events_cache: Arc<RwLock<HashMap<EventKey, K8sEvent>>>,
```

Comment:
```suggestion
    cache: Arc<RwLock<HashMap<EventKey, K8sEvent>>>,
```
minor preference nit: only one cache here

Reply:
Fixed in bebbb103967e201d7a393ae1865f4e21bcd72c42